src/sage/combinat/partition.py

```diff
@@ -6606,28 +6610,28 @@ def cardinality(self, algorithm='flint'):
             [[5], [4, 1], [3, 2], [3, 1, 1], [2, 2, 1], [2, 1, 1, 1], [1, 1, 1, 1, 1]]
             sage: len(v)
             7
-            sage: Partitions(5).cardinality(algorithm='gap')
+            sage: Partitions(5).cardinality(algorithm='gap')                            # needs sage.libs.gap
             7
-            sage: Partitions(5).cardinality(algorithm='pari')
+            sage: Partitions(5).cardinality(algorithm='pari')                           # needs sage.libs.pari
             7
-            sage: number_of_partitions(5, algorithm='flint')
+            sage: number_of_partitions(5, algorithm='flint')                            # needs sage.libs.flint
             7
 
         ::
 
-            sage: Partitions(10).cardinality()
+            sage: Partitions(10).cardinality()                                          # needs sage.libs.flint
             42
-            sage: Partitions(3).cardinality()
+            sage: Partitions(3).cardinality()                                           # needs sage.libs.flint
             3
-            sage: Partitions(10).cardinality()
+            sage: Partitions(10).cardinality()                                          # needs sage.libs.flint
```

Comment:
This doctest is repeated. 

May be we should reorder the doctests to use more block scope.

Reply:
Thanks, done in 51928653d51